kernel/x86_64/srot.c

```diff
@@ -0,0 +1,139 @@
+#include "common.h"
+
+#if defined(SKYLAKEX)
+#include "srot_microk_skylakex-2.c"
+#elif defined(HASWELL)
+#include "srot_microk_haswell-2.c"
+#endif
+
+#ifndef HAVE_SROT_KERNEL
+
+static void srot_kernel(BLASLONG n, FLOAT *x, FLOAT *y, FLOAT c, FLOAT s)
+{
+    BLASLONG i = 0;
+    FLOAT f0, f1, f2, f3;
+    FLOAT x0, x1, x2, x3;
+    FLOAT g0, g1, g2, g3;
+    FLOAT y0, y1, y2, y3;
+
+    FLOAT* xp = x;
+    FLOAT* yp = y;
+
+    BLASLONG n1 = n & (~7);
```

Comment:
n1 = n & (~3) ?

Reply:
This won't cause error result. May be not good for performance when n < 8.

Reply:
Thx, I was curious about that your step was 4 int the loop but ~7 was taken.

Reply:
I reference it from https://github.com/xianyi/OpenBLAS/blob/7f11e33e8d3c979183a4785c9d4519555ccb1bba/kernel/power/drot.c#L105

I'm not sure which one is good. If n1 = n & (~3) is better, I'll fix this code.

Reply:
Thanks for setting me straight. No need to fix.